flashlight/fl/tensor/backend/jit/JitTensorBase.cpp

```diff
@@ -23,6 +23,15 @@ JitTensorBase::~JitTensorBase() {
   node_->decRefCount();
 }
 
+// take care of refcount for old & new nodes
+void JitTensorBase::replaceNode(Node* newNode) const {
+  if (node_ != newNode) {
+    node_->decRefCount();
+    newNode->incRefCount();
+    const_cast<JitTensorBase*>(this)->node_ = newNode;
```

Comment:
Should this really be const/is it called from `const` members on `JitTensorBase`?

Reply:
Yes, the call chain is

`JitTensorBackend::eval --> JitTensorBase::eval --> JitTensorBase::replaceNode`.

As discussed in earlier comment, the best solution would be making `TensorBackend::eval` take in non-const `Tensor`. I'll send another PR soon.

Reply:
Sounds good, although it's probably a lot better if `fl::eval` still took a const Tensor — `const_cast` seems a bit ugly. Maybe we can discuss on the PR — thanks for looking into that!